src/WaveTrack.h

```diff
@@ -623,14 +623,23 @@ class AUDACITY_DLL_API WaveTrackFactory final
    const SampleBlockFactoryPtr &GetSampleBlockFactory() const
    { return mpFactory; }
 
+   /**
+    * \brief Creates an empty WaveTrack with default sample format and default rate
+    * \return Orphaned WaveTrack
+    */
+   std::shared_ptr<WaveTrack> Create();
+
+   /**
+    * \brief Creates an empty WaveTrack with custom sample format and custom rate
+    * \param format Desired sample format
+    * \param rate Desired sample rate
+    * \return Orphaned WaveTrack
+    */
+   std::shared_ptr<WaveTrack> Create(sampleFormat format, double rate);
```

Comment:
I understand that you changed the factory to force you to fix compilation of all uses of it, and so consider at each place how to set the track name.

How can we guarantee that future calls to these functions will also have names set?

Should you at least comment here that the caller should consider setting a unique track name?

Reply:
This function doesn't set the name. The reason is that the track with random name isn't better than track without the name, when the intention is to have a track with a certain name. Another reason is that there are many cases when track is created to hold some temprorary data and then inserted into another track. I'll improve the comment by adding "an unnamed empty track".